### ISSUE TITLE
multi source support mangadex

### ISSUE THREAD

**Issue by ikhoeyZX** [keiyoushi/tachiyomi-extensions-source#6235]
### Source name

mangadex

### Source language

all

### Describe your suggested feature

allow extension change main / alt server (like batoto does)
mangadex have 2 source
main : mangadex.org
dev : canary.mangadex.dev

### Other details

*some country have issue when access mangadex.org because blocked by goverment

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/mihonapp/mihon/issues/new/choose).
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

### 

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/keiyoushi/extensions-source/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Eagle1337** [comment]
Does the Dex have other official mirrors though?

**Comment by ikhoeyZX** [comment]
> Does the Dex have other official mirrors though?

yep
canary.mangadex.dev is early access feature for main domain 
it can be accessed by everyone (so basically it's official mirror domain)
![Screenshot_20241124_230700_Discord](https://github.com/user-attachments/assets/76e70c06-f9d3-45e6-aeba-1b23915c8b31)

**Comment by Norsze** [comment]
:/ seems like a pain. Point of an early access front end is that they can change fast without disturbing regular users. From a development side that just sounds like a source that can change the theme and code very suddenly to test stuff, which will break regularly without warning. But I'll let the maintainers decide

**Comment by AwkwardPeak7** [comment]
yeah no, we don't wanna use canary api in the extension
